File: Daily Test/C++/Next Non-Visited City 17.1.2026.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    int R, C, x, y;
    cin >> R >> C >> x >> y;
    string S;
    cin >> S;

    set<pair<int,int>> visited;
    visited.insert({x, y});

    for (char d : S) {
        if (d == 'N') {
            x = max(1, x - 1);
            while (visited.count({x, y}) && x > 1) x--;
        }
        if (d == 'S') {
            x = min(R, x + 1);
            while (visited.count({x, y}) && x < R) x++;
        }
        if (d == 'E') {
            y = min(C, y + 1);
            while (visited.count({x, y}) && y < C) y++;
        }
        if (d == 'W') {
            y = max(1, y - 1);
            while (visited.count({x, y}) && y > 1) y--;
        }
        visited.insert({x, y});
    }

    cout << x << " " << y;
}

File: Daily Test/Python/Next Non-Visited City 17.1.2026.py
R, C, X, Y = map(int, input().split())
S = input().strip()

x, y = X, Y
visited = set()
visited.add((x, y))

for d in S:
    if d == 'N':
        x = max(1, x - 1)
        while (x, y) in visited and x > 1:
            x -= 1
    elif d == 'S':
        x = min(R, x + 1)
        while (x, y) in visited and x < R:
            x += 1
    elif d == 'E':
        y = min(C, y + 1)
        while (x, y) in visited and y < C:
            y += 1
    elif d == 'W':
        y = max(1, y - 1)
        while (x, y) in visited and y > 1:
            y -= 1

    visited.add((x, y))

print(x, y)